src/ORM/Rule/ExistsIn.php

```diff
@@ -98,6 +109,11 @@ public function __invoke(EntityInterface $entity, array $options)
             return true;
         }
 
+        if ($this->_options['allowSqlNulls'] === true
+            && $this->_checkPartialSchemaNulls($entity, $source) === true
+        ) {
+            return true;
```

Comment:
No. But I want to come there when I become able to speak/listen English.

And I understand what you want. As I always use a surrogate key, I didn't know how NULL values in a composite foreign key are handled in SQL. I thought that MySQL/PostgreSQL rejects to insert a composite key like the above ( `[ parent_id => null, site_id => 99999 ]` ), 
but they actually allow to insert any values if it contains partially NULL.

The current `_fieldsAreNull` behavior seems incompatible with MySQL and PostgreSQL at least, when a composite foreign key allows NULL values. Personally I think you can fix the current behavior instead of adding a new option. (Because incompatible with common databases)